pkg/cassandra/config_tag.go

```diff
@@ -0,0 +1,97 @@
+package cassandra
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Masterminds/semver/v3"
+	"strings"
+)
+
+const (
+	cassConfigTagName             = "cass-config"
+	cassConfigTagOptionRecurse    = "recurse"
+	cassConfigTagOptionRetainZero = "retainzero"
+)
+
+// cassConfigTag is the representation of a cass-config tag contents.
+type cassConfigTag struct {
+	paths      []cassConfigTagPath
+	retainZero bool
+	recurse    bool
+}
+
+func (t cassConfigTag) pathForVersion(version *semver.Version) *cassConfigTagPath {
+	for _, path := range t.paths {
+		if valid, _ := path.constraint.Validate(version); valid {
+			return &path
+		}
+	}
+	return nil
+}
+
+type cassConfigTagPath struct {
+	constraint *semver.Constraints
+	path       string
+	segments   []string
+}
+
+func (p cassConfigTagPath) isInline() bool {
+	return len(p.segments) == 0
+}
+
+// parseCassConfigTag parses a textual representation of a cass-config tag.
+// The textual representation is of the following general form:
+// tag := <path1>[,<path2>,...,<pathN>][;<option1>;<option2>;...;<optionN>]
+// path := <constraint>:[<segments>]
+// constraint := a valid semver constraint
+// segments := a series of path segments separated by a slash, e.g. foo/bar/qix
+// option := recurse | retainzero
+func parseCassConfigTag(text string) (*cassConfigTag, error) {
```

Comment:
I think it will be rather complex to use regex here, especially to parse version constraints. The full syntax for version contraints is explained [here](https://github.com/Masterminds/semver#basic-comparisons). If we really want something more robust, I'd rather advocate for a grammar engine like ANTLR.

Reply:
Sounds good 👍